cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -2568,84 +2568,92 @@ namespace {
 
 std::mutex aws_init_lock;
 Aws::SDKOptions aws_options;
-std::atomic<bool> aws_initialized(false);
+bool aws_initialized = false;
 
-Status DoInitializeS3(const S3GlobalOptions& options) {
-  Aws::Utils::Logging::LogLevel aws_log_level;
+struct AwsInstance : public ::arrow::internal::Executor::Resource {
+  explicit AwsInstance(const S3GlobalOptions& options) {
+    Aws::Utils::Logging::LogLevel aws_log_level;
 
 #define LOG_LEVEL_CASE(level_name)                             \
   case S3LogLevel::level_name:                                 \
     aws_log_level = Aws::Utils::Logging::LogLevel::level_name; \
     break;
 
-  switch (options.log_level) {
-    LOG_LEVEL_CASE(Fatal)
-    LOG_LEVEL_CASE(Error)
-    LOG_LEVEL_CASE(Warn)
-    LOG_LEVEL_CASE(Info)
-    LOG_LEVEL_CASE(Debug)
-    LOG_LEVEL_CASE(Trace)
-    default:
-      aws_log_level = Aws::Utils::Logging::LogLevel::Off;
-  }
+    switch (options.log_level) {
+      LOG_LEVEL_CASE(Fatal)
+      LOG_LEVEL_CASE(Error)
+      LOG_LEVEL_CASE(Warn)
+      LOG_LEVEL_CASE(Info)
+      LOG_LEVEL_CASE(Debug)
+      LOG_LEVEL_CASE(Trace)
+      default:
+        aws_log_level = Aws::Utils::Logging::LogLevel::Off;
+    }
```

Comment:
is the new tab an expected change?

Reply:
Yes.  The initialize and finalize methods were file-scoped methods previously.  Now they have become the constructor/destructor of the `AwsInstance` class.  Since they are in a class they get some indentation.